streamwebs_frontend/streamwebs/views.py

```diff
@@ -24,11 +24,203 @@
     TransectZone, Canopy_Cover, CameraPoint, PhotoPoint,
     PhotoPointImage, Soil_Survey, Resource, School)
 
+from djqscsv import render_to_csv_response
+import csv
 import json
 import copy
 import datetime
 
 
+def export_wq(request, site_slug):
+    site = Site.objects.get(site_slug=site_slug)
+    waterq = Water_Quality.objects.filter(site_id=site.id)
+
+    # Store nid of each wq data sheet into a list
+    sheets = []
+    for each in waterq:
+        sheets.append(each.nid)
+
+    # Query the first set of samples outside of the loop and pop nid from list
+    samples = WQ_Sample.objects.prefetch_related('water_quality')
```

Comment:
In the loop, I'm merging querysets with ``samples = samples | n_samples``. I need to fetch the first data sheet outside of the loop in order to create a union with the following querysheets later.

Is there a more intelligent way to do this that I'm missing?